base/libgit2/oid.jl

```diff
@@ -155,17 +155,14 @@ function GitShortHash(obj::GitObject)
     return sid
 end
 
-Base.hex(id::GitHash) = join([hex(i,2) for i in id.val])
-Base.hex(id::GitShortHash) = hex(id.hash)[1:id.len]
-
 """
     raw(id::GitHash) -> Vector{UInt8}
 
 Obtain the raw bytes of the [`GitHash`](@ref) as a vector of length $OID_RAWSZ.
 """
 raw(id::GitHash) = collect(id.val)
 
-Base.string(id::AbstractGitHash) = hex(id)
+Base.string(id::AbstractGitHash) = base(16, id)
```

Comment:
Is there a `base(::Int, ::AbstractGitHash)` method defined somewhere?

What I meant in my previous comment was to define:
```
Base.string(id::GitHash) = join([base(i,2,pad=2) for i in id.val])
Base.string(id::GitShortHash) = string(id.hash)[1:id.len]
```

Reply:
Oh got it I thought you meant these were already defined

Reply:
This should actually be a method of `print`, then you also get `string` for free.

Reply:
We usually define `string` when it's actually easier to compute the String representation than to print it. This is quite rare, but if it can be implemented by calling `base`, this would be one of those cases.